native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import SimpleMarkdown from 'simple-markdown';
+
+type State = { [string]: any };
+
+type ASTNode = SingleASTNode | Array<SingleASTNode>;
+
+type SingleASTNode = {
+  type: string,
+  [string]: any,
+};
+
+type ReactElement = React$Element<any>;
+
+type Output<Result> = (node: ASTNode, state?: ?State) => Result;
+
+export default function(styles: any, opts: any = {}) {
+  const LINK_INSIDE = '(?:\\[[^\\]]*\\]|[^\\]]|\\](?=[^\\[]*\\]))*';
+  const LINK_HREF_AND_TITLE =
+    '\\s*<?([^\\s]*?)>?(?:\\s+[\'"]([\\s\\S]*?)[\'"])?\\s*';
+  var pressHandler = function(target) {
+    if (opts.onLink) {
+      opts.onLink(target).catch(function(error) {
+        console.log(
+          'There has been a problem with this action. ' + error.message,
+        );
+        throw error;
+      });
+    }
+  };
+  return {
+    autolink: {
+      react: function(
+        node: SingleASTNode,
+        output: Output<ReactElement>,
+        state: State,
+      ) {
+        state.withinText = true;
+        const _pressHandler = () => {
+          pressHandler(node.target);
+        };
+        const autolink = React.createElement(
```

Comment:
We should prefer JSX syntax (which gets transpiled to this) since it's more readable.

(I'm mentioning this here once, but please make sure to apply it everywhere in the PR!)